source/n_fields_problem.cc

```
@@ -444,6 +443,10 @@ template <int dim, int spacedim, int n_components, typename LAC>
 void NFieldsProblem<dim, spacedim, n_components, LAC>::refine_mesh ()
 {
   computing_timer.enter_section ("   Mesh refinement");
+
+  typedef TrilinosWrappers::MPI::BlockVector pVEC;
+  typedef BlockVector<double> sVEC;
+
   if (adaptive_refinement)
```

Qui bisognerebbe lasciare il compito all'interfaccia di decidere su quali componenti fare lo stimatore... 

chiamare 
~~~
interface->estimate_error_per_cell(...);
~~~
con la default che chiama kelly error estimator su una mask passata da parametri, come qui sotto ma non con la ComponentMask() data per scontata...